### PR TITLE
Fix to Date/Time Pickers crashing from no content

### PR DESCRIPTION
**Pull request by NicolasChampagne**
GitHub Issue (If applicable): #

## PR Type
What kind of change does this PR introduce?
Bugfix
Refactor

## What is the current behavior?
App may crash on clicking flyout buttons

## What is the new behavior?
Flyout button are working

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

Internal Issue (If applicable):
https://nventive.visualstudio.com/Umbrella/_workitems/edit/156490

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.iOS.cs [hunk 1]
```diff
@@ -3,16 +3,16 @@
 using Uno.Disposables;
 using Uno.UI.Common;
 using Uno.UI.DataBinding;
+using Windows.UI.Xaml.Controls.Primitives;
 using Windows.UI.Xaml.Data;
 
 namespace Windows.UI.Xaml.Controls
 {
-	public partial class TimePickerFlyout : Flyout
+	public partial class TimePickerFlyout : FlyoutBase
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.iOS.cs]:
```suggestion
	public partial class TimePickerFlyout : PickerFlyoutBase
```

File: src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePickerFlyout.iOS.cs [hunk 2]
```diff
@@ -47,6 +48,37 @@ private void InitializeContent()
 			AttachAcceptCommand((DatePickerSelector)Content);
 		}
 
+		#region Content DependencyProperty
+		public IUIElement Content
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePickerFlyout.iOS.cs]:
```suggestion
		internal IUIElement Content
```

File: src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePickerFlyout.iOS.cs [hunk 3]
```diff
@@ -47,6 +48,37 @@ private void InitializeContent()
...
+		{
+			get { return (IUIElement)this.GetValue(ContentProperty); }
+			set { this.SetValue(ContentProperty, value); }
+		}
+
+		public static readonly DependencyProperty ContentProperty =
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePickerFlyout.iOS.cs]:
```suggestion
		internal static readonly DependencyProperty ContentProperty =
```

File: src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.iOS.cs [hunk 4]
```diff
@@ -54,6 +54,38 @@ private void InitializeContent()
 			this.Binding(nameof(ClockIdentifier), nameof(ClockIdentifier), Content, BindingMode.TwoWay);
 		}
 
+		#region Content DependencyProperty
+		public IUIElement Content
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.iOS.cs]:
```suggestion
		internal IUIElement Content
```

File: src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.iOS.cs [hunk 5]
```diff
@@ -54,6 +54,38 @@ private void InitializeContent()
...
+		{
+
+			get { return (IUIElement)this.GetValue(ContentProperty); }
+			set { this.SetValue(ContentProperty, value); }
+		}
+
+		public static readonly DependencyProperty ContentProperty =
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.iOS.cs]:
```suggestion
		internal static readonly DependencyProperty ContentProperty =
```

File: src/Uno.UI/UI/Xaml/Controls/Primitives/PickerFlyoutBase.cs [hunk 6]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Windows.UI.Xaml.Controls.Primitives
+{
+	public partial class PickerFlyoutBase : FlyoutBase
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/Primitives/PickerFlyoutBase.cs]: Should have a protected ctor

File: src/Uno.UI/Uno.UI.csproj [hunk 7]
```diff
@@ -337,6 +337,10 @@
 		</ProjectReference>
 	</ItemGroup>
 
+	<ItemGroup>
+	  <UpToDateCheckInput Remove="UI\Xaml\Controls\Primitives\PickerFlyoutBase.cs" />
```
Comment (davidjohnoliver) [inline src/Uno.UI/Uno.UI.csproj]: undo this change

File: src/Uno.UI/UI/Xaml/Controls/Primitives/PickerFlyoutBase.cs (hunk 6 above)
Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/Controls/Primitives/PickerFlyoutBase.cs]: The _constructor_ should be protected, the class itself must be public.

Comment (NicolasChampagne) [reply src/Uno.UI/UI/Xaml/Controls/Primitives/PickerFlyoutBase.cs]: Like this?

Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/Controls/Primitives/PickerFlyoutBase.cs]: That's the one :)
